java-checks/src/main/java/org/sonar/java/checks/DoublePrefixOperatorCheck.java

```diff
@@ -38,35 +41,52 @@
     return ImmutableList.of(Tree.Kind.LOGICAL_COMPLEMENT, Tree.Kind.BITWISE_COMPLEMENT, Tree.Kind.UNARY_PLUS, Tree.Kind.UNARY_MINUS);
   }
 
-  static HashSet<ExpressionTree> prefixSet = new HashSet<>();
+  private Set<ExpressionTree> prefixSet = new HashSet<>();
+
+  @Override
+  public void scanFile(JavaFileScannerContext context) {
+    prefixSet.clear();
+    super.scanFile(context);
+  }
 
   @Override
   public void visitNode(Tree tree) {
-    UnaryExpressionTree temp = (UnaryExpressionTree) tree;
-    if (usedForDoublePrefix(temp)) {
+    UnaryExpressionTree exprTree = (UnaryExpressionTree) tree;
+    if (alreadyReported(exprTree)) {
       return;
     }
-    ExpressionTree expr = ExpressionUtils.skipParentheses(temp.expression());
-    if (temp.is(expr.kind())) {
+    ExpressionTree expr = ExpressionUtils.skipParentheses(exprTree.expression());
+    if (caseHandlingDoubleTilde(expr)) {
```

Comment:
This should be called from within the `exprTree.is(expr.kind())` condition (so only starting line 63), or you can not guarantee that both parent and child node are bitwise operator. You **need** to be sure that both parent and child are the same kind. On line 59, you don't know yet.